runtime/tests/integration_tests.rs

```diff
@@ -139,6 +140,109 @@ fn test_verify_cmd() {
+                pub_key_y: test_group.key.affine_y.as_slice()[..].try_into().unwrap(),
+                signature_r: Signature::from_der(&test.sig)
+                    .unwrap()
+                    .r()
+                    .to_bytes()
+                    .into(),
+                signature_s: Signature::from_der(&test.sig)
+                    .unwrap()
+                    .s()
+                    .to_bytes()
+                    .into(),
+                // Do tests on mailbox
+            };
+            let checksum = caliptra_common::checksum::calc_checksum(
+                u32::from(CommandId::ECDSA384_VERIFY),
+                &cmd.as_bytes()[4..],
+            );
+            let cmd = EcdsaVerifyCmd {
+                chksum: checksum,
+                ..cmd
+            };
+            fn push_err(err: &mut Vec<WycheproofResults>, id: usize, comment: &String) {
+                err.push(WycheproofResults {
+                    id: id,
+                    comment: comment.clone(),
+                });
+            }
+            match model.mailbox_execute(u32::from(CommandId::ECDSA384_VERIFY), cmd.as_bytes()) {
+                Ok(resp) => match test.result {
+                    wycheproof::TestResult::Valid | wycheproof::TestResult::Acceptable => {
+                        if !resp.is_none() || model.soc_ifc().cptra_fw_error_non_fatal().read() != 0
+                        {
+                            push_err(&mut wyche_err, test.tc_id, &test.comment);
+                        }
+                    }
+                    wycheproof::TestResult::Invalid => {
+                        push_err(&mut wyche_err, test.tc_id, &test.comment);
+                    }
+                },
+                Err(resp) => match test.result {
+                    wycheproof::TestResult::Valid | wycheproof::TestResult::Acceptable => {
+                        push_err(&mut wyche_err, test.tc_id, &test.comment);
+                    }
+                    wycheproof::TestResult::Invalid => {
+                        if let ModelError::MailboxCmdFailed(code) = resp {
+                            if code
+                                != Into::<u32>::into(
+                                    caliptra_drivers::CaliptraError::RUNTIME_ECDSA_VERIFY_FAILED,
+                                )
+                            {
+                                push_err(&mut wyche_err, test.tc_id, &test.comment);
+                                continue;
+                            }
+                        }
+                        if model.soc_ifc().cptra_fw_error_non_fatal().read()
+                            != Into::<u32>::into(
+                                caliptra_drivers::CaliptraError::RUNTIME_ECDSA_VERIFY_FAILED,
+                            )
+                        {
+                            push_err(&mut wyche_err, test.tc_id, &test.comment);
```

Comment:
What does this check? Are there some FW errors this should throw, such as if it apparently succeeds? If the registers may contain an error at this point, it should be cleared.

Reply:
I copied the checksum error which also does this. I think that error is never reported as a fw error indeed.

Reply:
Done